C++/test/data_structure/linked_list/singly_linked_list.cpp

```diff
@@ -0,0 +1,46 @@
+#include "third_party/catch.hpp"
+#include "data_structure/linked_list/singly_linked_list.hpp"
+
+TEST_CASE("Create an empty linked list", "[singly-linked-list]") {
+	SinglyLinkedList<int> list;
+	REQUIRE(list.is_empty() == true);
+}
+
+TEST_CASE("Add elements", "[singly-linked-list]") {
+	SinglyLinkedList<int> list;
+	list.insert_rear(2);
+	list.insert_rear(3);
```

Comment:
None of the tests cases singly linked list make use of `insert_front()`, `operator[]` or `is_empty()`. Calling these functions in various unit tests will improve code coverage.